x-pack/plugins/triggers_actions_ui/public/application/components/update_api_key_modal_confirmation.tsx

```diff
@@ -25,7 +26,8 @@ export const UpdateApiKeyModalConfirmation = ({
   onSearchPopulate,
 }: {
   onCancel: () => void;
-  idsToUpdate: string[];
+  idsToUpdate?: string[];
+  rulesToUpdate?: RuleTableItem[];
```

Comment:
Why do you need both `idsToUpdate` and `rulesToUpdate`? Can you pass `rulesToUpdate.map((rule) => rule.id` from rules list component as well? So it will be the same as from rules details. Then we can keep only `idsToUpdate` as a param.

Reply:
++

Reply:
Testing it, it looks like ids is always empty but we might be missing a use case, @JiaweiWu ?

Reply:
The reason why it still exists is because `rule_details` also uses this component. To reduce the amount of changes we need to do for this PR, both were left in.

We can have a comment here to refactor the `rule_details` implementation